core-bundle/src/DependencyInjection/Configuration.php

```diff
@@ -377,8 +377,50 @@ static function (array $value): array {
                     ->defaultNull()
                 ->end()
                 ->arrayNode('valid_extensions')
+                    ->info('Adds, removes or overwrites the list of enabled image extensions that can be used.')
                     ->prototype('scalar')->end()
                     ->defaultValue(['jpg', 'jpeg', 'gif', 'png', 'tif', 'tiff', 'bmp', 'svg', 'svgz', 'webp', 'avif'])
+                    ->example(['+heic', '-svgz', 'jxl'])
+                    ->validate()
+                        ->ifTrue(
+                            static function (array $extensions): bool {
+                                foreach ($extensions as $extension) {
+                                    if (!preg_match('/^[+-]?.+$/', $extension)) {
+                                        return true;
+                                    }
+                                }
+
+                                return false;
+                            },
+                        )
+                        ->thenInvalid('Make sure your provided image extensions are valid and optionally start with +/- to add/remove the extension to/from the default list.')
+                    ->end()
+                    ->validate()
+                        ->always(
+                            static function (array $extensions): array {
+                                $newList = array_filter($extensions, static fn ($extension) => !\in_array($extension[0], ['-', '+'], true));
+
+                                if ([] !== $newList) {
+                                    return array_unique($newList);
```

Comment:
Shouldn‘t `['+heic', '-svgz', 'jxl']` be turned into `['jxl', 'heic']` instead of returning just `['jxl']` here?

Reply:
It's simply the example values that needs an update since `['+heic', '-svgz', 'jxl']` would always just return `['jxl']` as the only valid image extension.

`$newList` actually just removes every `+` and `-` and is an early return if a normal extension has been found.

It would never make sense to use `['+heic', '-svgz', 'jxl']` but it would make sense to use `['+heic', '-svgz']`

Reply:
@ausi WDYT?

Reply:
I updated it here but let's discuss what example should be best:
https://github.com/contao/contao/pull/7478/commits/15430cb8a89b22bd6987f067577a7a0e152b1abc

Reply:
We should use the same algorithm as we do here: https://github.com/contao/contao/blob/49c56bd953ec3b05bd91b45df9c835d9b374fb00/core-bundle/src/Intl/Locales.php#L151-L166

`['+heic', '-svgz', 'jxl']` should result in `['jxl', 'heic']` then.

It might make sense to move this algorithm to the `ArrayUtil` as we use it here too: https://github.com/contao/contao/blob/49c56bd953ec3b05bd91b45df9c835d9b374fb00/core-bundle/src/Intl/Countries.php#L77-L92